src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs

```diff
@@ -42,6 +43,15 @@ public DynamicObservableObjectMetaObject(Expression parameter, DynamicObservable
         }
 
         /// <summary>
+        /// Returns the enumeration of all dynamic member names.
+        /// </summary>
+        /// <returns>The list of dynamic member names.</returns>
+        public override IEnumerable<string> GetDynamicMemberNames()
+        {
+            return ((DynamicObservableObject)Value).GetPropertyNames();
```

Comment:
No need to safe-cast here? I am not sure, I don't have the full code in my head.